File: java/order-api/src/main/java/br/com/brunosansp/order/model/Order.java
package br.com.brunosansp.order.model;

import java.time.LocalDateTime;
import java.util.List;

public class Order {
    private Long id;
    private String customerId;
    private List<OrderItem> items;
    private LocalDateTime createdAt;
    
    public Order() {
    }
    
    public Order(Long id, String customerId, List<OrderItem> items) {
        this.id = id;
        this.customerId = customerId;
        this.items = items;
        this.createdAt = LocalDateTime.now();
    }
    
    public Long getId() {
        return id;
    }
    
    public void setId(Long id) {
        this.id = id;
    }
    
    public String getCustomerId() {
        return customerId;
    }
    
    public void setCustomerId(String customerId) {
        this.customerId = customerId;
    }
    
    public List<OrderItem> getItems() {
        return items;
    }
    
    public void setItems(List<OrderItem> items) {
        this.items = items;
    }
    
    public LocalDateTime getCreatedAt() {
        return createdAt;
    }
    
    public void setCreatedAt(LocalDateTime createdAt) {
        this.createdAt = createdAt;
    }
}

File: go/order-api/internal/models/order.go
package models

import "time"

type Order struct {
	ID		 	int64
	CustomerID 	string
	Items    	[]OrderItem
	CreatedAt 	time.Time
}

type OrderItem struct {
	ID			int64
	ProductID 	string
	Quantity  	int
}

func NewOrder(id int64, customerID string, items []OrderItem) *Order {
	return &Order{
		ID:		 id,
		CustomerID: customerID,
		Items:    items,
		CreatedAt: time.Now(),
	}
}
